### PR TITLE
Implemented Book Detail with Availability Status and Comments

### PR DESCRIPTION
**Pull request by SalvadorCarnelutti**
## Summary ##

**book-detail**: Implemented book detail to show:
- Title
- Availability Status
- Author
- Year Released
- Genre

In said view one can also see the users comments of the book, while displaying:
- Username
- User Profile Picture
- Comment

Also made connection between the book detail view and the previous library view so it can be jumped between a specific book and the whole array of book presented.

A brief description of the pull request

## Screenshots ##
image

image

## Known issues ##

Can't seem to shrink detail view long titles even when modifying from the Attributes Inspector.

### CODE REVIEW

File: WBooks/AppDelegate.swift [hunk 1]
```diff
@@ -28,8 +28,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         window = UIWindow(frame: UIScreen.main.bounds)
         //Desde donde empieza la aplicacion.
         window?.rootViewController = TabBarController()
+        //window?.rootViewController = UINavigationController(rootViewController: BookDetailViewController())
```
Comment (acid-wolox) [inline WBooks/AppDelegate.swift]: If you're not using this line in the future, remove it.

File: WBooks/Views/Book Detail/BookDetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        _view.lblStatus.text = _viewModel.status.capitalized
+
+        _view.lblAuthor.text = _viewModel.author
+        _view.lblYear.text = _viewModel.year
+        _view.lblGenre.text = _viewModel.genre
+        let imageUrl = URL(string: _viewModel.image)
+        let data = try? Data(contentsOf: imageUrl ?? "http://wolox-training.s3.amazonaws.com/uploads/6942334-M.jpg")
+        if let imageData = data {
+            let bookImage = UIImage(data: imageData)
+            _view.imgBookCover.image = bookImage
```
Comment (acid-wolox) [inline WBooks/Views/Book Detail/BookDetailViewController.swift]: Is there a reason why you split this into two lines? I think you can do:
`_view.imgBookCover.image = UIImage(data: imageData)`

File: WBooks/Views/Book Detail/BookDetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(cell: BookDetailCell.self)!
+        let commentView = _bookDetailViewModel.comments[indexPath.row]
+        cell.lblUser.text = commentView.user.username
+        cell.lblComment.text = commentView.content
+        let imageUrl = URL(string: commentView.user.image)
+        let data = try? Data(contentsOf: imageUrl ?? "https://goo.gl/1PBWVM")
+        if let imageData = data {
+            let bookImage = UIImage(data: imageData)
+            cell.imageUser.image = bookImage
```
Comment (acid-wolox) [inline WBooks/Views/Book Detail/BookDetailViewController.swift]: Is there a reason why you split this into two lines? I think you can do:
`cell.imageUser.image = UIImage(data: imageData)`

Comment (acid-wolox) [reply WBooks/Views/Book Detail/BookDetailViewController.swift]: Also, I'd reorder and give some space to the syntax:
```
        let cell = tableView.dequeue(cell: BookDetailCell.self)!
        let commentView = _bookDetailViewModel.comments[indexPath.row]
        let imageUrl = URL(string: commentView.user.image)
        let data = try? Data(contentsOf: imageUrl ?? "https://goo.gl/1PBWVM")

        cell.lblUser.text = commentView.user.username
        cell.lblComment.text = commentView.content

        if let imageData = data {
            cell.imageUser.image = UIImage(data: imageData
        }
```

File: WBooks/Views/Book Detail/BookDetailViewController.swift (hunk 2 above)
Comment (acid-wolox) [reply WBooks/Views/Book Detail/BookDetailViewController.swift]: Same as the other one, I'd swap the order of some lines and give some space:

```
        let imageUrl = URL(string: _viewModel.image)
        let data = try? Data(contentsOf: imageUrl ?? "http://wolox-training.s3.amazonaws.com/uploads/6942334-M.jpg")

        _view.lblBookTitle.text = _viewModel.title
        _view.lblStatus.text = _viewModel.status.capitalized
        _view.lblAuthor.text = _viewModel.author
        _view.lblYear.text = _viewModel.year
        _view.lblGenre.text = _viewModel.genre

        if let imageData = data {
            _view.imgBookCover.image = UIImage(data: imageData)
        }
```

File: WBooks/Views/Library/LibraryViewController.swift [hunk 4]
```diff
@@ -84,4 +74,33 @@ extension LibraryViewController: UITableViewDataSource, UITableViewDelegate {
...
         return cell
     }
+    
+    //Cuando se seleccione alguna de las columas, redirigir de vista.
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        // Guarda el book instance del libro de la fila seleccionada.
+        let book = _viewModel.books[indexPath.row]
+        //********let bookDetailViewModel = BookDetailViewModel(bookVM: book)
+        //********let bookDetailViewController = BookDetailViewController(viewModel: bookDetailViewModel)
+        //********let bookDetailViewController = BookDetailViewController(viewModel: book)
```
Comment (acid-wolox) [inline WBooks/Views/Library/LibraryViewController.swift]: These comments shouldn't be in the final version for a PR, as they already were committed to the branch previously and will remain there.

**Review by acid-wolox — CHANGES_REQUESTED** [review]
Awesome work here, Salvador! There are some comments about minor things, about style, mainly.

File: WBooks/Views/Book Detail/BookDetailViewController.swift (hunk 2 above)
Comment (SalvadorCarnelutti) [reply WBooks/Views/Book Detail/BookDetailViewController.swift]: When you tell me to give some space, what do you mean?

**Review by acid-wolox — APPROVED** [review]
Great work!

**Pull request merged**